deepchecks/base/check.py

```diff
@@ -346,9 +346,9 @@ def run(self, model) -> CheckResult:
         pass
 
 
-@dataclass
 class CheckFailure:
     """Class which holds a run exception of a check."""
 
-    check: Any  # Check class, not instance
-    exception: Exception
+    def __init__(self, check: Any, exception: Exception):
+        self.check = check
+        self.exception = exception
```

Comment:
@matanper do we need a dataclass here?
it broke the linkcode from sphinx, so I did this workaround. it might be better to add a workaround in docs/source/conf.py, but for now is this ok?